Assets/Scripts/Interacting.cs

```diff
@@ -0,0 +1,62 @@
+using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+using UnityEngine.UI;
+
+public class Interacting : MonoBehaviour
+{
+    [SerializeField] private Image centerDot;
+    [SerializeField] private float pickingDistanse = 2f;
+    [SerializeField] private PlayerHands playerHands;
+    private Ray ray;
+    
+    void Update()
+    {
+        CheckHit();
+    }
+
+    private void CheckHit()
+    {
+        ray = Camera.main.ViewportPointToRay(new Vector3(0.5f, 0.5f, 0f));
+        if (Physics.Raycast(ray, out RaycastHit hit) & hit.collider.tag == "CanInteract" & hit.distance <= pickingDistanse) 
```

Comment:
Unity has tags but using them is kina bad idea, because it is based on string, maintaining and debugging it is a nightmare.
Instant of it you can check if object has component that you are looking for f.e TorchStand by TryGetComponent.
In this case the best way is to check if object is interactable so.
Create interface in .cs file
```cs
public interface IInteractable { // in C# we start name of interface witch I
    void Interact(GameObject interacter);
}
```
And now you can make simple logic in interacting.cs
``` cs
if (Physics.Raycast(ray, out RaycastHit hit) && hit.distance <= pickingDistanse && hit.collider.TryGetComponent(out IInteractable interactableObject)) 
{
   centerDot.enabled = true;
   if (Input.GetKeyDown(KeyCode.Mouse0))
   {
      interactableObject.Interact(this);
   }
}
else
{
    centerDot.enabled = false;
}
```
And thats it.
It is solid because this script is use to interact witch any Interactable object it should not be responsible for each interaction